### PR TITLE
qa/tasks: add positive test for ceph_volume_client method

### PR DESCRIPTION
**Pull request by rishabh-d-dave**
`ceph_volume_client.py's` `put_object_versioned()` has only one test. Since
this only test is a negative test it fails to assure that
`put_object_versioned()` works as expected with positive inputs even when
it completes successfully. Therefore, write a positive test for better
coverage.

### CODE REVIEW

**Comment by rishabh-d-dave** [comment]
@batrick, the context for this PR is written nowhere except for the commit message and the email I had sent you. Should I create a tracker issue for this?

**Comment by rishabh-d-dave** [comment]
This PR depends on one of the commits on PR #27718. Adding a DNM for that reason.

**Comment by rishabh-d-dave** [comment]
@batrick can you please to this comment -https://github.com/ceph/ceph/pull/27805#issuecomment-486947929?

File: qa/tasks/cephfs/test_volume_client.py [hunk 1]
```diff
@@ -988,6 +988,26 @@ def test_put_object_versioned(self):
...
         self._configure_vc_auth(vc_mount, "manila")
 
+        obj_data = 'test_data'
+        obj_name = 'test_vc_obj'
+        pool_name = self.fs.get_data_pool_names()[0]
+        self.fs.rados(['put', obj_name, '-'], pool=pool_name, stdin_data=obj_data)
+
+        self._volume_client_python(vc_mount, dedent("""
+                data, version_before = vc.get_object_and_version("{pool_name}", "{obj_name}")
+                data += 'modification1'
```
Comment (batrick) [inline qa/tasks/cephfs/test_volume_client.py]:
````
2019-05-15T17:17:08.304 INFO:teuthology.orchestra.run.smithi019:Running:
2019-05-15T17:17:08.305 INFO:teuthology.orchestra.run.smithi019:> sudo adjust-ulimits daemon-helper kill python3 -c '
2019-05-15T17:17:08.305 INFO:teuthology.orchestra.run.smithi019:> from __future__ import print_function
2019-05-15T17:17:08.305 INFO:teuthology.orchestra.run.smithi019:> from ceph_volume_client import CephFSVolumeClient, VolumePath
2019-05-15T17:17:08.305 INFO:teuthology.orchestra.run.smithi019:> import logging
2019-05-15T17:17:08.306 INFO:teuthology.orchestra.run.smithi019:> from rados import OSError as rados_OSError
2019-05-15T17:17:08.306 INFO:teuthology.orchestra.run.smithi019:> log = logging.getLogger("ceph_volume_client")
2019-05-15T17:17:08.306 INFO:teuthology.orchestra.run.smithi019:> log.addHandler(logging.StreamHandler())
2019-05-15T17:17:08.306 INFO:teuthology.orchestra.run.smithi019:> log.setLevel(logging.DEBUG)
2019-05-15T17:17:08.306 INFO:teuthology.orchestra.run.smithi019:> vc = CephFSVolumeClient("manila", "/etc/ceph/ceph.conf", "ceph", None, None)
2019-05-15T17:17:08.307 INFO:teuthology.orchestra.run.smithi019:> vc.connect()
2019-05-15T17:17:08.307 INFO:teuthology.orchestra.run.smithi019:>
2019-05-15T17:17:08.307 INFO:teuthology.orchestra.run.smithi019:> data, version_before = vc.get_object_and_version("cephfs_data", "test_vc_obj")
2019-05-15T17:17:08.307 INFO:teuthology.orchestra.run.smithi019:> data += '"'"'modification1'"'"'
2019-05-15T17:17:08.308 INFO:teuthology.orchestra.run.smithi019:> vc.put_object_versioned("cephfs_data", "test_vc_obj", data, version)
2019-05-15T17:17:08.308 INFO:teuthology.orchestra.run.smithi019:> data, version_after = vc.get_object_and_version("cephfs_data", "test_vc_obj")
2019-05-15T17:17:08.308 INFO:teuthology.orchestra.run.smithi019:> assert version_after == version_before + 1
2019-05-15T17:17:08.308 INFO:teuthology.orchestra.run.smithi019:>
2019-05-15T17:17:08.308 INFO:teuthology.orchestra.run.smithi019:> vc.disconnect()
2019-05-15T17:17:08.309 INFO:teuthology.orchestra.run.smithi019:>         '
2019-05-15T17:17:08.485 INFO:teuthology.orchestra.run.smithi019.stderr:Configuring to RADOS with config /etc/ceph/ceph.conf...
2019-05-15T17:17:08.496 INFO:teuthology.orchestra.run.smithi019.stderr:Connecting to RADOS...
2019-05-15T17:17:08.509 INFO:teuthology.orchestra.run.smithi019.stderr:Connection to RADOS complete
2019-05-15T17:17:08.509 INFO:teuthology.orchestra.run.smithi019.stderr:Connecting to cephfs...
2019-05-15T17:17:08.509 INFO:teuthology.orchestra.run.smithi019.stderr:CephFS initializing...
2019-05-15T17:17:08.514 INFO:teuthology.orchestra.run.smithi019.stderr:CephFS mounting...
2019-05-15T17:17:08.524 INFO:teuthology.orchestra.run.smithi019.stderr:Connection to cephfs complete
2019-05-15T17:17:08.524 INFO:teuthology.orchestra.run.smithi019.stderr:Recovering from partial auth updates (if any)...
2019-05-15T17:17:08.532 INFO:teuthology.orchestra.run.smithi019.stderr:Nothing to recover. No auth meta files.
2019-05-15T17:17:08.533 INFO:teuthology.orchestra.run.smithi019.stderr:Traceback (most recent call last):
2019-05-15T17:17:08.533 INFO:teuthology.orchestra.run.smithi019.stderr:  File "<string>", line 13, in <module>
2019-05-15T17:17:08.533 INFO:teuthology.orchestra.run.smithi019.stderr:TypeError: can't concat bytes to str
2019-05-15T17:17:08.535 INFO:teuthology.orchestra.run.smithi019.stderr:disconnect
2019-05-15T17:17:08.535 INFO:teuthology.orchestra.run.smithi019.stderr:Disconnecting cephfs...
2019-05-15T17:17:08.535 INFO:teuthology.orchestra.run.smithi019.stderr:Disconnecting cephfs complete
2019-05-15T17:17:08.535 INFO:teuthology.orchestra.run.smithi019.stderr:Disconnecting rados...
2019-05-15T17:17:08.536 INFO:teuthology.orchestra.run.smithi019.stderr:Disconnecting rados complete
2019-05-15T17:17:08.582 INFO:teuthology.orchestra.run.smithi019.stderr:daemon-helper: command failed with exit status 1
2019-05-15T17:17:08.588 DEBUG:teuthology.orchestra.run:got remote process result: 1
2019-05-15T17:17:08.590 INFO:tasks.cephfs_test_runner:test_put_object_versioned (tasks.cephfs.test_volume_client.TestVolumeClient) ... ERROR
````

From: /ceph/teuthology-archive/pdonnell-2019-05-15_13:28:11-fs-wip-pdonnell-testing-20190515.015737-distro-basic-smithi/3957014/teuthology.log

I think the above line is actually line 13 for the traceback. Should this then be `b'modification1'`?

**Comment by rishabh-d-dave** [comment]
Closing in favour of #28692; moving my commits there.

**Pull request closed without merging**